hazelcast/src/test/java/com/hazelcast/client/protocol/compatibility/ReferenceObjects.java

```diff
@@ -782,7 +785,9 @@ private static boolean isEqualStackTrace(StackTraceElement stackTraceElement1, S
             aBoolean, aBoolean, aBoolean, aBoolean, aHotRestartConfig, anEventJournalConfig, aString, aListOfData,
             aMergePolicyConfig, aBoolean, aListOfListenerConfigHolders);
     private static MemberVersion aMemberVersion = new MemberVersion(aByte, aByte, aByte);
-    public static Collection<MemberInfo> aListOfMemberInfos = Collections.singletonList(new MemberInfo(anAddress, aUUID, aMapOfStringToString, aBoolean, aMemberVersion));
+    public static Collection<MemberInfo> aListOfMemberInfos = Collections.singletonList(new MemberInfo(anAddress, aUUID, aMapOfStringToString, aBoolean, aMemberVersion,
+            ImmutableMap.of(EndpointQualifier.MEMBER, anAddress)));
```

Comment:
Can you change the ordinal value to 2 at the protocol side to use `ProtocolType.WAN` ? Because the binary encoder is encoding `localhost` as the identifier string for the `EndpointQualifier`. I see that we can specify a custom identifier using `EndpointQualifier#resolve` with the protocol type of `WAN` but it is not an option for other protocol types. 

We can configure binary generator to encode custom values for fix sized types(as you did with ordinal), but there is no way to do so for var sized types in the protocol code generator. I think we should also provide an option for var sized types. I will create an issue for that.

Reply:
> Can you change the ordinal value to 2 at the protocol side to use ProtocolType.WAN ?

Great idea, thank you, this helped a lot.

> but there is no way to do so for var sized types

yep, I found the same. I think in `binary/util.py`, `VarSizedParamEncoder#encode_var_sized_frame()` has to be improved to do something similar to `CustomTypeEncoder#create_initial_frame()` (reading from `reference_objects.objects`). To do it, I think `custom_type_name` has to be passed to `#encode_var_sized_frame()` but that method is called from an other place too. I didn't dig deeper today.